### PR TITLE
fix: Dequeue items from RequestQueue in the correct order

### PR DESCRIPTION
**Pull request by janbuchar**

### CODE REVIEW

File: src/crawlee/storages/request_queue.py [hunk 1]
```diff
@@ -97,7 +96,7 @@ def __init__(
         self._internal_timeout_seconds = 5 * 60
         self._assumed_total_count = 0
         self._assumed_handled_count = 0
-        self._queue_head_dict: OrderedDictType[str, str] = OrderedDict()
+        self._queue_head_dict: OrderedDict[str, str] = OrderedDict()
```
Comment (B4nan) [inline src/crawlee/storages/request_queue.py:99]: so this line is the actual fix? it feels like you only changed the types, i thought that cannot have a runtime effect (just like with TS)?

or is it about the removed unknown label above?

Comment (janbuchar) [reply src/crawlee/storages/request_queue.py:99]: 'tis but a draft... I just fixed a bit of weirdness I noticed in the file. Sadly the actual fix still evades me.

File: src/crawlee/storages/request_queue.py [hunk 2]
```diff
@@ -472,7 +471,7 @@ async def get_handled_count(self) -> int:
     async def get_total_count(self) -> int:
         return self._assumed_total_count
 
-    async def ensure_head_is_non_empty(
+    async def _ensure_head_is_non_empty(
```
[the remark this answers is not in the document]
Comment (janbuchar) [reply src/crawlee/storages/request_queue.py:474]: I know, but I'd consider it a bug for this to be public.

File: src/crawlee/storages/request_queue.py [hunk 3]
```diff
@@ -31,6 +30,38 @@
 
 logger = getLogger(__name__)
 
+__all__ = ['RequestQueue']
+
+
+T = TypeVar('T')
+
+
+class BoundedSet(Generic[T]):
```
Comment (B4nan) [inline src/crawlee/storages/request_queue.py:39]: maybe add a comment on why this is needed

**Review by B4nan — APPROVED** [review]
not that i would understand what this is doing completely 🙃

**Pull request merged**